### PR TITLE
Refactor CMakeLists and fix build errors

### PR DESCRIPTION
**Pull request by waifod**
## Summary

We reorganise the CMakeLists file to make it more readable and manageable.

Secondly, we add some imports to fix build errors we encountered.

## Testing

Executed the following commands:
```
cd ~/projects/em/
cmake ./
cmake --build .
./test/EmberScriptSnapshotTestRunner
```

Output:
```
/Users/dumatte/projects/em/test/snapshot/EmLang/snapshots/set_union: OK!
/Users/dumatte/projects/em/test/snapshot/EmLang/snapshots/function: OK!
/Users/dumatte/projects/em/test/snapshot/EmLang/snapshots/set_intersection: OK!
/Users/dumatte/projects/em/test/snapshot/EmLang/snapshots/virtual_set: OK!
/Users/dumatte/projects/em/test/snapshot/EmLang/snapshots/set_equality: OK!
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,72 +1,24 @@
 cmake_minimum_required(VERSION 3.28)
-project(em)
+project(em
+  VERSION 1.0
+  DESCRIPTION "EmberScript - A small language for mathematical operations"
```
Comment (lluiscamino) [inline CMakeLists.txt]: Not [EmberScript](https://github.com/ghempton/ember-script)! :(

**Comment by lluiscamino** [comment]
Thanks! Probably we should add build instructions to the README or create a script as well. For now, I was just using the Build button in CLion

File: CMakeLists.txt (hunk 1 above)
Comment (waifod) [reply CMakeLists.txt]: ooops, fixing xd

**Comment by waifod** [comment]
Alright, I'll get around to updating the readme.

File: CMakeLists.txt (hunk 1 above)
Comment (waifod) [reply CMakeLists.txt]: Addressed. Thanks for pointing it out.

**Pull request merged**